### PR TITLE
Release: 6.0.0rc0

### PR DESCRIPTION
**Pull request by mdickinson**
Prepare for 6.0.0 release candidate:

- Add release date to changelog
- Add support for a prerelease section in the setup.py machinery
- Bump the version for the release.

### CODE REVIEW

**Comment by mdickinson** [comment]
Found a test suite bug; let's get that fixed first.

**Comment by mdickinson** [comment]
> Found a test suite bug; let's get that fixed first.

I was sure I'd posted a follow-on comment to this, but apparently I didn't submit it. That bug was #847. It was fixed in master in #848, and that fix was backported to the release branch in #850.

This PR is ready for review, except that the release date is wrong. I'll update it.

File: setup.py [hunk 1]
```diff
@@ -22,16 +22,17 @@
 MAJOR = 6
 MINOR = 0
 MICRO = 0
-IS_RELEASED = False
+PRERELEASE = "rc0"
```
Comment (corranwebster) [inline setup.py]: For non-prerelease is this going to be `""`?  Do these changes to setup.py need to be pushed across to master (or should the changes to setup.py be applied to master and then backported)?

Comment (mdickinson) [reply setup.py]: > For non-prerelease is this going to be `""`?

Yes.

> Do these changes to setup.py need to be pushed across to master

Yes. There's an entry in the release checklist with a reminder to do this.

Comment (mdickinson) [reply setup.py]: I cherry-picked the middle commit across to master in #864.

**Pull request merged**